soc/xtensa/intel_adsp/tools/hw-service.py

```diff
@@ -0,0 +1,449 @@
+
+                    os.system(f"sudo kill -9 {proc.pid} ")
+                return
+
+            time.sleep(1)
+
+class device_runner():
+    def __init__(self, board, serial, load_cmd, lock_enable):
+        self.fw_file = None
+        self.lock = threading.Lock()
+        self.fw_queue = queue.Queue()
+
+        # Use for device mutual exclusion
+        self.lock_enable = lock_enable
+        self.available = True
+        self.occpied_user = None
+        self.passcode = None  # Not use yet
+
+        # Board specific config
+        self.board = board_config(board, serial, load_cmd)
+        self.load_cmd = self.board.get_cmd()
+        self.is_load_fw = self.board.is_load_fw()
+
+    def set_dev_lock(self, user, passcode):
+        if not self.lock_enable:
+            return True
+
+        if not self.available:
+            return False
+
+        self.lock.acquire()
+        self.available = False
+        self.occpied_user = user
+        self.passcode = passcode
+        self.lock.release()
+        return True
+
+    def set_dev_unlock(self, user, passcode):
+        if self.lock_enable:
+            self.lock.acquire()
+            self.available = True
+            self.occpied_user = None
+            self.passcode = None
+            self.lock.release()
+
+    def is_dev_available(self, user):
+        if not self.lock_enable:
+            return True
+
+        if self.occpied_user == user or self.available:
+            return True
+        else:
+            return False
+
+    def set_fw_ready(self, fw_recv):
+        if fw_recv:
+            self.fw_queue.put(fw_recv)
+
+    def is_fw_ready(self):
+        self.fw_file = self.fw_queue.get()
+        log.info(f"Current FW is {self.fw_file}")
+
+        if self.fw_file:
+            if self.fw_file == "OTHER":
+                self.fw_file = None
+            return True
+        else:
+            return False
+
+    def cleanup(self):
+        self.lock.acquire()
+        self.script = None
+        if self.fw_file:
+            os.remove(self.fw_file)
+        self.fw_file = None
+        self.lock.release()
+
+    def get_script(self):
+        if os.geteuid() != 0:
+            self.script = ([f'sudo', f'{self.load_cmd}'])
+        else:
+            self.script = ([f'{self.load_cmd}'])
+
+        if self.is_load_fw:
+            self.script.append(f'{self.fw_file}')
+
+        if self.board.params:
+            for param in self.board.params:
+                self.script.append(param)
+
+        log.info(f'run script: {self.script}')
+        return self.script
+
+class board_config():
+    def __init__(self, board, serial, load_cmd):
+
+        board_intel_adsp = ['intel_adsp_cavs15', 'intel_adsp_cavs18',
+                            'intel_adsp_cavs20', 'intel_adsp_cavs20_jsl',
+                            'intel_adsp_cavs25', 'intel_adsp_cavs25_tgph']
+
+        board_x86_hypervisor = ['acrn_ehl_crb', 'acrn']
```

Comment:
isn't it cavs specific ?

Reply:
En...I made it also support other boards, but it seems not good to put them in the this PR/commit will put them in another PR . Thanks for the review!

Reply:
it is still here ... if this file is supposed to be used for other targest other than intel_adsp, it has to be moved to a better place

Reply:
Ok, got it. @ceolin , yes, I made this intel_adsp only now.